osutil/statelock.go

```diff
@@ -0,0 +1,93 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//go:build statelock
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package osutil
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"time"
+)
+
+func traceCallers(description string) {
```

Comment:
```suggestion
func traceCallers(description string) error {
```
and then you can update MaybeSave to do:
```
	if heldMs > threshold || waitMs > threshold {
		formattedLine := fmt.Sprintf("lock: held %d ms wait %d ms", heldMs, waitMs)
		if err := traceCallers(formattedLine); err != {
			logger.Infof("cannot trace callers: %v", err)
		}
	}
```
and change `fmt.Fprintf()` to use `return fmt.Errorf("cannot this or that: %w", err)`

Reply:
I tried that initially bet realized that we cannot use logger in utils, is there any other way to use it?

Reply:
I implemented using fmt.Fprintf() because I saw faultinject does that too.